### DISCUSSION TITLE
Home Assistant Roborock Integration - Selective Room Cleaning

### DISCUSSION THREAD

**Discussion by incomputerate** [humbertogontijo/homeassistant-roborock#636] [Q&A]
Hi, I am using "app_segment_clean" in Roborock Integration to clean specific rooms in the order I want - Room ID 17, 16, 18, 
When the robot finishes room 17, I want to send the robot to a specific spot of room 17 (using Roborock: Vacuum go to and x-y coordinates), pause for a frew seconds, and then resume the selective room cleaning (i.e., proceed with room 16). 

I wrote an automation with UI and do not know what command/service to achieve what I want. I tried using Timer and set its duration to 15 minutes (the time the robot usually takes to clean room 17) right after the selective room cleaning starts.  I then tried using wait_for_trigger to detect if the Timer is finished, then send to robot to the parkign spot with Go To service call. But after selective room cleaning started, the robot immediately moved to the paking spot in room 17 and just sit there without cleaning room 17. I used Edit YAML option and copied and pasted YAML codes below.

Automation
alias: S7 – Selective Room Cleaning
description: ""
trigger: []
condition: []
action:
  - service: roborock.vacuum_goto
    target:
      entity_id: vacuum.rr_s7
    data:
      x_coord: "22736"
      y_coord: "29181"
  - delay:
      hours: 0
      minutes: 0
      seconds: 40
      milliseconds: 0
  - service: roborock.vacuum_clean_segment
    target:
      entity_id: vacuum.rr_s7
    data:
      segments:
        - 17
        - 16
        - 18
      repeats: 1
  - service: timer.start
    target:
      entity_id: timer.timer_park_hallway
    data:
      duration: "00:15:00"
  - wait_for_trigger:
      - platform: state
        entity_id:
          - timer.timer_park_hallway
        to: idle
  - service: roborock.vacuum_goto
    target:
      entity_id: vacuum.rr_s7
    data:
      x_coord: "22582"
      y_coord: "32148"
  - delay:
      hours: 0
      minutes: 0
      seconds: 10
      milliseconds: 0
mode: single

Is there way to achieve what I want – parking the robot at certain spot at the end of room 17 cleaning, pause for 10 seconds, and resume the selective room cleaning?

**Comment by Choleriker** [comment]
Where do you have the app segments coordinates from? Could you please share more about your solution, I want to do similar thingy.

I assume your problem here is that your robot triggers the segment clean, but doesnt really give you a response when segment cleaning was done. It just adds a queue what to clean. So after first segment start, just in time, your waiter fires. Have you tried to run the segment cleaning, and then wait until the robot says: "Go home"? That is then the point to trigger the "Stop" command, wait, and start from scratch for the next segment. Just as an idea.

**Comment by incomputerate** [comment]
I’ve put the auto door opener/closer and auto ramp car projects on hold for a few months. They’re a bit too challenging for my age, especially since I don’t have much background in IT or networking.

As for Home Assistant, I installed the Xiaomi Vacuum Map Card Lovelace after setting up the Roborock Integration. After configuring the map card for a specific Roborock robot, I noticed that the dropdown menu in the card allows me to use the "Pin & Go" option. When I select a spot on the map and click "Edit map," the line labeled “Selection” would typically display the X & Y coordinates for that position.

However, I’ve recently encountered a problem. The “Selection” line is no longer showing any values. This issue started after I removed and then reinstalled the Roborock Integration. After adding the vacuum map cards again, I suspect I didn’t configure them properly, leading to this malfunction.

But you can see the documentation in the link below. In this case, the author used Zone (instead of Pin & Go) option and the line "selection" produces the zone coordinates. But the concept is the same. 






 <https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/blob/master/docs/templates/setup.md#getting-coordinates> lovelace-xiaomi-vacuum-map-card/docs/templates/setup.md at master · PiotrMachowski/lovelace-xiaomi-vacuum-map-card

 <https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/blob/master/docs/templates/setup.md#getting-coordinates> github.com

 

If you go to the home page of the above link, there are instructions on how to install the map card.

 

To execute the Go To, I wrote a script, selected the Action “Other Action” -> “Roborock” -> “Roborock Vacuum go to” and specified the device/entity and the x-y coordinate.
